matt-mike/public/scripts/article.js

```diff
@@ -0,0 +1,150 @@
+  };
+
+  // DONE: Chain together a `map` and a `reduce` call to produce an array of unique author names. You will
+  // probably need to use the optional accumulator argument in your reduce call.
+  Article.allAuthors = () => {
+    return Article.all.map(article => article.author)
+                      .reduce(function(allAuthors, authorName) {
+                        if(allAuthors.indexOf(authorName) < 0) allAuthors.push(authorName);
+                        return allAuthors;
+                      }, [])
+  };
+
+  Article.numWordsByAuthor = () => {
+    return Article.allAuthors().map(author => {
+      // DONE: Transform each author string into an object with properties for
+      // the author's name, as well as the total number of words across all articles
+      // written by the specified author.
+      // HINT: This .map should be setup to return an object literal with two properties.
+      // The first property should be pretty straightforward, but you will need to chain
+      // some combination of filter, map, and reduce to get the value for the second
+      // property.
+      return {author: `${author}`,
+        wordCount: Article.all.filter(function(a) {
+          return a.author === author
+        }).map(function(a){
+          return a.body.split(' ').length
+        }).reduce(function(aggregator, value) {
+          return aggregator + value
+        })
+      }
+    })
+  };
+
+  Article.truncateTable = callback => {
+    $.ajax({
+      url: '/articles',
+      method: 'DELETE',
+    })
+    .then(console.log) // REVIEW: Check out this clean syntax for just passing 'assumed' data into a named function!
+                       // The reason we can do this has to do with the way Promise.prototype.then works. It's a little
+                       // outside the scope of 301 material, but feel free to research!
+    .then(callback);
+  };
+
+  Article.prototype.insertRecord = function(callback) {
+    // REVIEW: Why can't we use an arrow function here for .insertRecord()??
+    $.post('/articles', {author: this.author, authorUrl: this.authorUrl, body: this.body, category: this.category, publishedOn: this.publishedOn, title: this.title})
+    .then(console.log)
+    .then(callback);
+  };
+
+  Article.prototype.deleteRecord = function(callback) {
+    $.ajax({
+      url: `/articles/${this.article_id}`,
+      method: 'DELETE'
+    })
+    .then(console.log)
+    .then(callback);
+  };
+
+  Article.prototype.updateRecord = function(callback) {
+    $.ajax({
+      url: `/articles/${this.article_id}`,
+      method: 'PUT',
+      data: {
+        author: this.author,
+        authorUrl: this.authorUrl,
+        body: this.body,
+        category: this.category,
+        publishedOn: this.publishedOn,
+        title: this.title,
+        author_id: this.author_id
+      }
+    })
+    .then(console.log)
+    .then(callback);
+  };
+  app.Article = Article;
```

Comment:
Since we are passing in `app` as the parameter `module`, you should attach `Article` to `module` instead of to `app`.